### PR TITLE
Fix the wrong mapping in sequential labels transform

### LINKED ISSUE #840: SequentialLabels generates a wrong mapping
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

> For example, if a label map has 6 labels with IDs (3, 5, 9, 15, 16, 23), then this will apply a RemapLabels transform with remapping={3: 1, 5: 2, 9: 3, 15: 4, 16: 5, 23: 6}. 

The documentation describes that the behavior of SequentialLabels is to map (3, 5, 9, 15, 16, 23) to (1, 2, 3, 4, 5, 6). But when I use SequentialLabels, I get (3, 1, 2, 3, 4, 5)

And I have read the source code, `SequentialLabels` uses the code below to generate mapping.

```python
unique_labels = torch.unique(image.data)
remapping = {
     unique_labels[i].item(): i
     for i in range(1, len(unique_labels))
}
```

When we input the (3, 5, 9, 15, 16, 23), the mapping is {5: 1, 9: 2, 15: 3, 16: 4, 23: 5}, so the 3 is reused.

### Code for reproduction

```python
import torch
import torchio

subject = torchio.Subject(label=torchio.LabelMap(tensor=torch.Tensor([[[[3, 5, 9, 15, 16, 23]]]])))
transform = torchio.SequentialLabels()
subject = transform(subject)
print(subject["label"][torchio.DATA])
```

### Actual outcome

tensor([[[[3., 1., 2., 3., 4., 5.]]]])

### Error messages

_No response_

### Expected outcome

tensor([[[[1., 2., 3., 4., 5., 6.]]]])

### System info

```shell
Platform:   Windows-10-10.0.19044-SP0
TorchIO:    0.18.74
PyTorch:    1.10.2+cu102
SimpleITK:  2.1.1 (ITK 5.2)
NumPy:      1.22.2
Python:     3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)]
```

### PR DESCRIPTION
**Pull request by iamSmallY**
Fixes #840.

**Description**
The current mapping will map (3, 5, 9, 15, 16, 23) to {5: 1, 9: 2, 15: 3, 16: 4, 23: 5}, which is inconsistent with the documentation. And it can pass the test because in the testing procedure, when the label is generated, a 0 label will be added to it, and the 0 is the minimum in the label. But when we use an image as a label, if the minimum number is not 0, the `SequentialLabels` will cause a fatal bug.

I have no idea whether this is a non-breaking change, Only when the minimum number in label is 0 can lead to the same behavior with the edited `SequentialLabels`, but usually the minimum number in label is 0.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [ ] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [x] In-line docstrings updated
- [x] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [x] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by romainVala** [comment]
Thanks for the PR
It looks good to me !

**Comment by fepegar** [comment]
Hi, @iamSmallY. Thanks for your contribution!

@efirdc, could you please take a look at this PR?

**Comment by fepegar** [comment]
As we have no news, I'm going to merge this but I might also refactor the label transforms so the label `0` is not handled more carefully.

**Comment by fepegar** [comment]
@all-contributors please add @iamSmallY for bug.

**Comment by fepegar** [comment]
Thanks for your contribution, @iamSmallY!
